### PR TITLE
[ENH] Allow HonestForest to sample higher than max_samples # of bootstraps

### PR DESCRIPTION
**Pull request by adam2392**
Supersedes #204 

Changes proposed in this pull request:
- build_hyppo_oob_forest and HonestForestClassifier should work together to make the predictions
- adds the S@S98 function from #192 

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Review by adam2392 — COMMENTED** [review]
- `build_hyppo_oob_forest` and `HonestForestClassifier` should work together to make the predictions

File: sktree/stats/utils.py [hunk 1]
```diff
@@ -65,13 +66,45 @@ def _cond_entropy(y_true: ArrayLike, y_pred_proba: ArrayLike) -> float:
...
     "mse": mean_squared_error,
     "mae": mean_absolute_error,
     "balanced_accuracy": balanced_accuracy_score,
     "auc": roc_auc_score,
     "mi": _mutual_information,
     "cond_entropy": _cond_entropy,
+    "sas98": _SAS98,
 }
 
 POSTERIOR_FUNCTIONS = ("mi", "auc", "cond_entropy")
```
Comment (YuxinB) [inline sktree/stats/utils.py]: Should add "sas98" at **POSTERIOR_FUNCTIONS**?

**Review by PSSF23 — COMMENTED** [review]
I don't recognize the SAS98 name, we have S@98? Also, the metric should be included in `POSITIVE_METRICS` and `POSTERIOR_FUNCTIONS`

**Comment by YuxinB** [comment]
So this is not in pip now right? Would you mind letting us know once it can be install with pip?

**Comment by adam2392** [comment]
Sure. It will be on pip once this is merged. But would prefer everyone's okay before doing so as releasing on pip takes a while of my time.

File: sktree/stats/utils.py (hunk 1 above)
Comment (adam2392) [reply sktree/stats/utils.py]: Done

**Comment by adam2392** [comment]
I'm going to merge this in for now for the sake of getting it onto pypi, but I cannot guarantee no mistakes and would highly suggest reviewing the code before running any exps.

**Pull request merged**